### PR TITLE
Fix resendwallettransactions assert failure if -walletbroadcast=0

### LINKED ISSUE #10981: resendwallettransactions asserts if walletbroadcast=0
There is an assert in `CWalletTx::RelayWalletTransaction` that gets hit.  All other code paths check `fBroadcastTransactions` before getting to `CWallet::ResendWalletTransactionsBefore` except the rpc call `resendwalletransactions`.

It's an easy fix, but we just need to decide whether to throw an error or return an empty array of rebroadcasted transaction ids.

Should be fixed for 0.15.

### PR DESCRIPTION
**Pull request by TheBlueMatt**
This fixes #10981 in my preferred way.

### CODE REVIEW

**Comment by laanwj** [comment]
utACK

**Comment by sipa** [comment]
utACK 48ff1822a007195640299f728d75b52e6d4a22a6

**Comment by promag** [comment]
Even though `resendwallettransactions` is for testing purpose, having an empty response indicates that there are no unconfirmed transactions, which can be incorrect. IMO it should return an error like `RPC_INVALID_REQUEST`.

**Comment by TheBlueMatt** [comment]
@promag seems reasonable, I added a commit to do so.

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2597,6 +2598,9 @@ UniValue resendwallettransactions(const JSONRPCRequest& request)
 
     LOCK2(cs_main, pwallet->cs_wallet);
 
+    if (pwallet->GetBroadcastTransactions())
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: Nit, curly braces.

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -1868,6 +1868,8 @@ std::vector<uint256> CWallet::ResendWalletTransactionsBefore(int64_t nTime, CCon
     std::vector<uint256> result;
 
     LOCK(cs_wallet);
+    if (!fBroadcastTransactions) return result;
```
Comment (promag) [inline src/wallet/wallet.cpp]: Remove and let it hit assert.

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -2589,6 +2589,7 @@ UniValue resendwallettransactions(const JSONRPCRequest& request)
             "Immediately re-broadcast unconfirmed wallet transactions to all peers.\n"
             "Intended only for testing; the wallet code periodically re-broadcasts\n"
             "automatically.\n"
+            "Returns an RPC error if -walletbroadcast is set to false.\n"
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: In the help you reference the app arg, but not in the error message. Make them consistent?

**Review by promag — COMMENTED** [review]
Concept ACK modulus comments.

**Comment by TheBlueMatt** [comment]
Fixed issues and squashed.

**Comment by dougstrong77** [comment]
Plz send me some where to learn this im confused and screwing up everything

**Comment by jnewbery** [comment]
Post-merge comment: application-level errors should not use `RPC_INVALID_REQUEST`. Fixed in #11002.

**Pull request merged**